Fixes #3307: Removes empty lessons section & title from single course page.

hansschuijff opened the pull request:
Fixes #3307 

### Changes proposed in this Pull Request

* Remove empty lessons html from single course page
* In the process moves actions 'sensei_single_course_lessons_before' and 'sensei_single_course_lessons_after' outside of the course-lessons section, so the section can be dependent on post having been found.
* In Sensei LMS the above actions seem only to be used for for setting up and resetting the query and some globals, so moving them outside the section should be possible.

### Testing instructions

* Works on single course page ( /course/{course-slug}/ ) and a page using the [sensei_course_page] shortcode.

hansschuijff committed:
Removed empty lessons section from single course.

hansschuijff committed:
Lint-errors processed

donnapep commented:
On unsupported themes (themes that don't explicitly declare support for Sensei LMS or for which we don't have a custom integration) like Divi, there's an empty `<header><h1>` element:

![Screen Shot 2020-10-14 at 1 43 55 PM](https://user-images.githubusercontent.com/1190420/96025593-535a4080-0e23-11eb-8058-b4cad7b8b9bf.jpg)

On supported themes like Storefront, that element contains the course title:

![Screen Shot 2020-10-14 at 1 45 39 PM](https://user-images.githubusercontent.com/1190420/96025771-9b796300-0e23-11eb-84f2-6cc6d57b8df5.jpg)

Does it seem possible to remove the empty `<header>` tag for unsupported themes as well?

hansschuijff commented:
@donnapep I'll see what I can do. It's been a while, so I have to recall first what I have done here. I'll let you know.

hansschuijff committed:
Stopped generating empty <header> and <h1> tags

Changed all the_title() methods
so it only renders a <header> or <h1> tag
when they are not empty.

hansschuijff committed:
Fixed lint errors.

hansschuijff committed:
Fixed lint errors.

hansschuijff committed:
Fixed lint errors.

hansschuijff commented:
@donnapep I've added a condition to the_title() so the header/h1 is only inserted when the title is not empty. Since I found the same problem was also present in the_title() for lessons, I have performed the same edit in all 4 the_title() methods (course, lesson, quiz and message). In all cases the header block is only inserted when there is content in it. It seems to work like this.
